src/test/java/seedu/address/logic/commands/ListTagsCommandTest.java

```diff
@@ -90,4 +101,49 @@ public void execute_invalidState_throwAssertionError() throws CommandException {
         model.setBattleState(BattleState.PRE_BATTLE);
         cmd.execute(model, new CommandHistory());
     }
+
+    @Test
+    public void execute_testCommandResult_success() {
+        ListTagsCommand cmd = new ListTagsCommand();
+
+        StringBuilder expectedOutput = new StringBuilder();
+        expectedOutput.append(ListTagsCommand.MESSAGE_SUCCESS)
+                .append(model.getHumanPlayer().getFleet().getAllTags().size())
+                .append(ListTagsCommand.TAGS_FOUND);
+
+        int counter = 0;
+        for (Tag tag : model.getHumanPlayer().getFleet().getAllTags()) {
+            expectedOutput.append(tag.getTagName());
+            if (++counter < model.getHumanPlayer().getFleet().getAllTags().size()) {
+                expectedOutput.append(", ");
+
+            }
+        }
+
+        CommandResult expectedCommandResult = new CommandResult(expectedOutput.toString(), false, false);
+        assertCommandSuccess(cmd, model, commandHistory, expectedCommandResult, expectedModel);
+
+    }
+
+    @Test
+    public void execute_testEmptyCommandResult_success() {
```

Comment:
Testing for an empty command result is something we have to account for. This is a good inclusion. Thanks Ian!